tests/chainer_tests/functions_tests/noise_tests/test_gumbel_softmax.py

```diff
@@ -0,0 +1,41 @@
+import unittest
+
+import numpy
+
+import chainer
+from chainer import cuda
+from chainer import functions
+from chainer import testing
+from chainer.testing import attr
+
+
+@testing.parameterize(*testing.product({
+    'shape': [(3, 2), ],
+}))
```

Comment:
Can you add other dtypes? `'dtype': [numpy.float16, numpy.float32, numpy.float64]`